src/rust/guya/template/src/template.rs

```diff
@@ -0,0 +1,230 @@
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::net::HttpMethod,
+	std::net::Request,
+	std::String,
+	std::{html::Node, ObjectRef, Vec},
+	Chapter, DeepLink, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
+	MangaViewer, Page,
+};
+use alloc::string::ToString;
+extern crate alloc;
+
+pub struct GuyaSiteData {
+	pub base_url: String,
+	pub nsfw: MangaContentRating,
+	pub language: String,
+}
+
+impl Default for GuyaSiteData {
+	fn default() -> GuyaSiteData {
+		GuyaSiteData {
+			base_url: String::new(),
+			nsfw: MangaContentRating::Safe,
+			language: String::from("en"),
+		}
+	}
+}
+
+pub fn get_manga_list(data: GuyaSiteData, filters: Vec<Filter>, _: i32) -> Result<MangaPageResult> {
+	let url = format!("{}/api/get_all_series/", &data.base_url);
+	let request = Request::new(url, HttpMethod::Get).header("User-Agent", "Aidoku");
+	let mut json = request.json()?.as_object()?;
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				let query = filter.value.as_string()?.read();
+				for manga in json.keys() {
+					let title = manga.as_string()?.read();
+					if !title.to_lowercase().contains(&query.to_lowercase()) {
+						json.remove(&title);
+					}
+				}
+			}
+			_ => continue,
+		}
+	}
+
+	let mut manga_arr: Vec<Manga> = Vec::new();
+	let mut titles: Vec<String> = json.keys().map(|k| k.as_string().unwrap().read()).collect();
+	titles.sort();
+	for title in titles {
+		let obj = match json.get(&title).as_object() {
+			Ok(obj) => obj,
+			Err(_) => continue,
+		};
+		let slug = obj.get("slug").as_string()?.read();
+		let cover = format!("{}{}", &data.base_url, obj.get("cover").as_string()?.read());
```

Comment:
it's best not to use `?` since that will return the entire function if it fails to unwrap. Instead, you can match it and provide a default value if it fails.